bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/profile/ScriptProfile.java

```diff
@@ -0,0 +1,164 @@
+
+import java.util.List;
+
+import javax.script.ScriptException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.core.ConfigParser;
+import org.openhab.core.thing.profiles.ProfileCallback;
+import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.thing.profiles.ProfileTypeUID;
+import org.openhab.core.thing.profiles.StateProfile;
+import org.openhab.core.transform.TransformationException;
+import org.openhab.core.transform.TransformationService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.TypeParser;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ScriptProfile} is generic profile for managing values with scripts
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class ScriptProfile implements StateProfile {
+
+    public static final String CONFIG_SCRIPT_LANGUAGE = "scriptLanguage";
+    public static final String CONFIG_TO_ITEM_SCRIPT = "toItemScript";
+    public static final String CONFIG_TO_HANDLER_SCRIPT = "toHandlerScript";
+
+    private final Logger logger = LoggerFactory.getLogger(ScriptProfile.class);
+
+    private final ProfileCallback callback;
+    private final TransformationService transformationService;
+
+    private final List<Class<? extends State>> acceptedDataTypes;
+    private final List<Class<? extends Command>> acceptedCommandTypes;
+    private final List<Class<? extends Command>> handlerAcceptedCommandTypes;
+
+    private final String scriptLanguage;
+    private final String toItemScript;
+    private final String toHandlerScript;
+
+    private final boolean isConfigured;
+
+    public ScriptProfile(ProfileCallback callback, ProfileContext profileContext,
+            TransformationService transformationService) {
+        this.callback = callback;
+        this.transformationService = transformationService;
+
+        this.acceptedCommandTypes = profileContext.getAcceptedCommandTypes();
+        this.acceptedDataTypes = profileContext.getAcceptedDataTypes();
+        this.handlerAcceptedCommandTypes = profileContext.getHandlerAcceptedCommandTypes();
+
+        this.scriptLanguage = ConfigParser.valueAsOrElse(profileContext.getConfiguration().get(CONFIG_SCRIPT_LANGUAGE),
+                String.class, "");
+        this.toItemScript = ConfigParser.valueAsOrElse(profileContext.getConfiguration().get(CONFIG_TO_ITEM_SCRIPT),
+                String.class, "");
+        this.toHandlerScript = ConfigParser
+                .valueAsOrElse(profileContext.getConfiguration().get(CONFIG_TO_HANDLER_SCRIPT), String.class, "");
+
+        if (scriptLanguage.isBlank()) {
+            logger.error("Script language is not defined. Profile will discard all states and commands.");
```

Comment:
`warn` should do here

Reply:
I used `error` because that is also used in the other profiles:

https://github.com/openhab/openhab-core/blob/cb38d1936006798d4eb0f71651bd1cb7c34bd158/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/profiles/SystemOffsetProfile.java#L53-L74

Reply:
ok